### PR TITLE
gather_candidates: unless local_addresses are specified in config, we…

### PR DESCRIPTION
**Pull request by bitflows**
… should not stop the entire ice candidate gathering process, just because we fail to gather one of them.

### CODE REVIEW

**Comment by bitflows** [comment]
Wasn't sure if I should keep the check in nice_agent_send() or not. Its probably not needed anymore, but will prevent a segfault if it should fire of some unforseen reasons.

File: agent/agent.c [hunk 1]
```diff
@@ -1557,8 +1557,11 @@ nice_agent_gather_candidates (NiceAgent * agent, guint stream_id)
...
               "%u/%u: Unable to gather host candidate for address %s",
               stream->id, component->id, ip);
-          ret = FALSE;
-          goto error;
+          if (configured_local_addresses){
+            /* The client specified these addresses, so treat this as an error! */
+            ret = FALSE;
+            goto error;
+          }
         }
```
Comment (johnbassett) [inline agent/agent.c:1566]: Needs an else clause here as if we fall through with udp_host_candidate == NULL the following code to get stun and turn candidates will segault trying to access sockptr

File: agent/agent.c [hunk 2]
```diff
@@ -1643,8 +1649,11 @@ nice_agent_gather_candidates (NiceAgent * agent, guint stream_id)
...
               "%u/%u: Unable to gather tcp-act host candidate for address %s",
               stream->id, component->id, ip);
-          ret = FALSE;
-          goto error;
+          if (configured_local_addresses){
+            /* The client specified these addresses, so treat this as an error! */
+            ret = FALSE;
+            goto error;
+          }
         }
```
Comment (johnbassett) [inline agent/agent.c:1658]: Same as for the UDP case above, needs an else clause to protect following code

File: agent/agent.c (hunk 1 above)
Comment (bitflows) [reply agent/agent.c:1566]: I feared that would happen initially, but I'm seetting borh turn and stun while testing, and it didn't crash. I'll take a second look!

Comment (bitflows) [reply agent/agent.c:1566]: Perhaps I wasnt in "full_mode" when testing and this occured.

**Pull request closed without merging**